official/utils/misc/model_helpers.py

```diff
@@ -53,3 +54,32 @@ def past_stop_threshold(stop_threshold, eval_metric):
     return True
 
   return False
+
+
+def generate_synthetic_data(
```

Comment:
Why does this implementation differ from the current synthetic data approach? And could you document why it is better than just the naive `.from_tensor_slices(...).repeat(...)`?

Reply:
I think I'm missing something. The implementation is the same as the current synthetic data approach, except that it allows the input/label shapes to be nested.

Reply:
I think I'm likely the one missing something.